### DISCUSSION TITLE
The output problem

### DISCUSSION THREAD

**Discussion by LianGong98** [NGEET/fates#928] [Q&A]
Hi, I‘m running a FATES single point simulation in CESM2.2.0 and the FATES version is sci.1.30.0_api.8.0.0. I chose two pfts _(broadleaf_evergreen_tropical_tree, broadleaf_hydrodecid_tropical_tree_), adjusted some parameters, and initialized with inventory. It could run successfully. However, some outputs always zero, such as AGB, Total biomass, Leaf biomass. Besides, the H2OSIO seems unreasonable, it was too dry in the beginning. 
![H2OSOI](https://user-images.githubusercontent.com/97998383/198197242-21addfc7-9068-4f2b-8a63-b16c3bf2ec9a.jpg)

It is very difficult for me. Is it the problem of parameters? or the Initialization? and what should I do ? 
Any advice would be greatly appreciated.

**Comment by rosiealice** [comment]
That does seem like a very low initial soil moisture? Did everything die of hydraulic failure? If you plot out the number of individual what happens? I wonder why the soil is starting so dry? I'm about to head out but it's worth trying to figure out if all the tree have died. What type of mortality they died from, and perhaps what H2OSOI we are initializing with here in the init_cold routines.

**Comment by rosiealice** [comment]
The mortality rates are output as

ih_m1_si_scpf
To
ih_m8_si_scpf

And (if they aren't labelled in the output), they are set here:

main/FatesHistoryInterfaceMod.F90
Around line 2561. 
(Writing this on a train on my iPad so the line permalinks don't work)

**Reply by LianGong98** [reply]
Hi ,thank you for reply. The mortality rates from M1_SCPF to M8_SCPF are really weird, all were zero.  And I initialized it with reference to https://fates-users-guide.readthedocs.io/en/latest/user/Model-Initialization-Modes.html 
_use_fates_ed_st3 = .true.
fates_inventory_ctrl_filename= ...._
but the water  in Patch file format (*.pss) is zero Is soil initialization come from the Patch file format (*.pss)?  or should provide the initialization file? If not , how should I Initialize the H2OSOI?

**Reply by rosiealice** [reply]
Oh, its in St3 mode... I guess then you have static structure, by definition, and so my hypothesized cause isn't going to be applicable...

**Reply by rosiealice** [reply]
Sorry i didnt catch that from your post! What is the GPPand LAI? That would tell you if the physiology is working and whether this is an issue with the outputs or if the system isn't working correctly st all....

**Comment by LianGong98** [comment]
I am so soorry ,I  attach the diagram of LAI and GPP now 
![TLAI](https://user-images.githubusercontent.com/97998383/198268296-aadd9cd1-79a3-4089-88a1-71a94d81fd8a.png)
![GPP](https://user-images.githubusercontent.com/97998383/198268326-e1f91094-27fd-4d94-95e1-4d62465d20dd.png)
and the GPP values like this :
![GPP](https://user-images.githubusercontent.com/97998383/198269139-cdff0d3c-6fe1-4208-9fa2-8ae205a0de1b.jpg)
and its units: gC/m^2/s

**Reply by rosiealice** [reply]
No worries! That looks to me like the model is running internally OK, but that it isn't printing our the biomass values etc. Perhaps because these are inputs not outputs in St3 mode, but I am not an expert on how that works Maybe @ckoven would know if we should be reporting those things?

You can just throw away the output until the soil moisture has stabilized here, as it doesn't look like it's having any major impact on the later outputs.

**Reply by LianGong98** [reply]
Thank you very much！I will try it and check my forcing data

**Comment by dlawrenncar** [comment]
We have seen something like this in the past with CLM if CO2 was
inadvertently set to 0.  Or, this could happen if there is something wrong
with the forcing data (e.g., solar radiation is 0).  Just a couple things
that are worth checking.

**Reply by LianGong98** [reply]
OK .That's a good advice ,I will check my  forcing data .Thank you very much!

**Reply by rosiealice** [reply]
I'm not sure anything is running incorrectly, those GPP numbers look fairly normal given their weird units.

**Reply by LianGong98** [reply]
Emmm.. maybe my incident solar radiationsomething wrong? but I am not sure:
![FSDS_in_2003-02](https://user-images.githubusercontent.com/97998383/198315030-a1563fe5-114d-4d51-8987-beddce71cac7.png)
![incident solar radiation](https://user-images.githubusercontent.com/97998383/198315098-a8eb0d52-61a1-4879-9e9d-ad06f864c3fe.jpg)

**Comment by dlawrenncar** [comment]
That looks right to me.  As Rosie said, GPP is actually not that low, so I
would agree with her perspective that this doesn't seem like a forcing
issue.

**Comment by rgknox** [comment]
@LianGong98 I saw that you are using the "st3" mode.  This does require an initial condition of ecosystem structure. It appears that you have LAI, so it looks like you successfully initialized.  Could you describe the initial condition, ie is it from census data, or something else?

**Reply by LianGong98** [reply]
Thank you very much for your reply ,the initial data  derived from census data, just like this 
![patch file](https://user-images.githubusercontent.com/97998383/198430458-c8239f02-0229-4885-8882-4daeb048e5f7.jpg)
I tried to change the value of water to 0.3, but it seemed  to have no effect on results

**Reply by mpaiao** [reply]
The water column is just a legacy column from the original ED model (which generated pss and css files as outputs).
I don't think this is used by FATES, actually I think the only columns that are read in by FATES are trk, age, and area.

**Comment by mpaiao** [comment]
Am I interpreting it wrong? It seems GPP is going to zero after the beginning of the simulation, and then bouncing back, so I don't think it's an output issue, it may be a problem with initial conditions...

A couple of thoughts based on previous runs that caused massive dieback in the tropics:
1. Long wave radiation. @LianGong98 if you are running at BCI, the meteorological forcing does not have incoming long wave radiation. If `FLDS` is not removed from `<path_to>/run/datm.streams.txt.CLM1PT.CLM_USRDAT`, the model runs with zero long wave radiation and you end up with an ice age that kills all the plants. I don't think this is your case because plants eventually grow, but it may be worth checking the long wave radiation output (Variable `FLDS`).
2. Soil moisture. When I run with hydro deciduous, I ended up editing the initial soil moisture in the host land model. In the case of CLM, I replaced `use_fates_planthydro` with `use_fates` in [this `if` block](https://github.com/ESCOMP/CTSM/blob/616905bbbac55f506fccc5aef8c45bbbdbba94a2/src/biogeophys/WaterStateType.F90#L351-L355) in biogeophys/WaterStateType.F90 (and also in the use statement [at the beginning of the file](https://github.com/ESCOMP/CTSM/blob/616905bbbac55f506fccc5aef8c45bbbdbba94a2/src/biogeophys/WaterStateType.F90#L17)). This had more of an effect on deciduous, not much in the evergreens.

**Reply by LianGong98** [reply]
yeah I did remove the FLDS 
_./preview_namelists
cp run/datm.streams.txt.CLM1PT.CLM_USRDAT user_datm.streams.txt.CLM1PT.CLM_USRDAT
`sed -i '/FLDS/d' user_datm.streams.txt.CLM1PT.CLM_USRDAT`_
but I'm not sure whether I need to run ./preview_namelists  again.
If I want to change the  initial soil moisture, shoud I  replaceduse_fates_planthydro with use_fates just like you did?

**Reply by mpaiao** [reply]
This is the correct setting for skipping long wave radiation from site met drivers, so you probably don't have the zero FLDS problem (worth checking the output just in case). 

Regarding the initial soil moisture, yes, just replace `use_fates` instead of `use_fates_planthydro`, this should increase the initial moisture for every FATES run (not only when FATES-Hydro is enabled).

**Reply by LianGong98** [reply]
Thank you so much for your assistance , I'll do it. 

 Best wishes

**Reply by LianGong98** [reply]
@mpaiao Hi, I'm sorry to bother you again, there was a new error in plant hydraulics when I ran 
![ERROR (2)](https://user-images.githubusercontent.com/97998383/198885770-4ef92569-43d7-4cf9-bb6e-33d189d42a45.jpg)
but this error disappeared when the hydraulic module was turned off (_**use_fates_planthydro = .false.**_  )
It's really annoying, I do not know why it happened. 
Any advice would be greatly appreciated !

**Comment by rosiealice** [comment]
I am still not totally sure what proble you ar trying to solve here.  As I see it, the soil moisture starts off too low, but then equilibrates and GPP seems to be normal after that. Because we are in st3 mode there isn't a feedback to the low productivity, and so I think you can just throw out this spinup phase output (it is normal practice to throw away the output before the soil moisture equilibrates in prescribed LAI mode(s)). 

Is there some other problem I'm missing?
